### ISSUE TITLE
my code cannot stop 

### ISSUE THREAD

**Issue by scq-wdx** [matplotlib/mplfinance#151]
When l use `mpf.plot()` to plot my picture, my code is continues running, without an In[1] into In[2] in the console.

    import pandas as pd
    import mplfinance as mpf

    df_csv=pd.read_csv("wanke reverse.csv",index_col=0,parse_dates=True)
    df_csv.index.name="Date"
    df_csv.rename(columns=
        {"open":"Open","high":"High","low":"Low","close":"Close","volume":"Volume"},
        inplace=True)

    mpf.plot(df_csv,type="candle",mav=8,volume=True)

**Comment by DanielGoldfarb** [comment]
I don't know what you mean by "my code is continues running, without an In[1] into In[2] in the console."

Can you provide a screen shot, and describe what you expect, and describe what you are seeing that is different from what you expect?

**Comment by AndySum** [comment]
How long did you leave it running for? Have you tried plotting with a smaller amount of data?

**Comment by moonmoondog** [comment]
Same issue happens to me when I select %matplotlib auto for interactive chart. My IDE is Spyder.

**Comment by DanielGoldfarb** [comment]
@scq-wdx @moonmoondog 

There is nothing we can do with such a vague description of your issue.  Please provide more exact details as requested above.  _Explain exactly how we can reproduce your issue._  Full environment details, including where and how you are running your code.  What you expect to see, and what you are seeing. Screen shots may be helpful.

This sounds very much like an environment issue (not matplotlib nor mplfinance related).  None-the-less we are happy to help if you can provide some specifics.

**Comment by moonmoondog** [comment]
> @scq-wdx @moonmoondog
> 
> There is nothing we can do with such a vague description of your issue. Please provide more exact details as requested above. _Explain exactly how we can reproduce your issue._ Full environment details, including where and how you are running your code. What you expect to see, and what you are seeing. Screen shots may be helpful.
> 
> This sounds very much like an environment issue (not matplotlib nor mplfinance related). None-the-less we are happy to help if you can provide some specifics.

Thank you for the reply. 
I am using Spyder 4.0.1 with Python 3.7.6, Qt5.12.7, PyQt5 5.12.3.
mplfinance version 0.12.5a2
matplotlib version 3.2.0
My code is just simply 
```
import mplfinance as mpf
mpf.plot(OHLC) 
```
(where OHLC is a pandas dataframe object with open, high, low, close, volume columns) 
When I tried to plot the mpf graph inline with the code above, there is no issues at all. 
But when I put in a magic command in the iPython platform 
`%matplotlib auto` 
or
`%matplotlib qt5`
for a pop-up interactive graph, a graph window popped up as expected. However the command line does not stop nor terminate by itself just like how the OP descibed. Tried CTRL-C and CTRL-Pause to get out of the process, but to no avial. I had to restart the entire console to make command input works again. 
PS: I have been using the vanila matplotlib with no problems. This problem seems to be only assoiciated with mplfinance. 

Thank you for this wonderful tool BTW :)

**Comment by DanielGoldfarb** [comment]
@moonmoondog 

try `block=False`:

    mpf.plot(OHLC,block=False)

Let me know if that works.

**Comment by moonmoondog** [comment]
> @moonmoondog
> 
> try `block=False`:
> 
> ```
> mpf.plot(OHLC,block=False)
> ```
> 
> Let me know if that works.

Thank you for the prompt reply! 
I have tried to add the parameter `block = False ` as suggested. 
It solved the command line frozen problem. However, the pop-up graph failed to populate and seemed to enter some kind of loop like this: 
![error](https://user-images.githubusercontent.com/66976593/84862314-870ba080-b0a5-11ea-8c49-5141a086ea75.jpg)
When I tried to force closing the window, the whole iPython kernel failed. 
Once again, it works fine with the inline option. It just acts weird when pop-up option is selected. 
I have also tried to run the pop-up graph in another thread. However, it freezes the command input as well when the thread re-joins the main.

**Comment by AminSaqi** [comment]
> block=False



> @moonmoondog
> 
> try `block=False`:
> 
> ```
> mpf.plot(OHLC,block=False)
> ```
> 
> Let me know if that works.

Fixed my issue.

**Comment by moonmoondog** [comment]
> @moonmoondog
> 
> try `block=False`:
> 
> ```
> mpf.plot(OHLC,block=False)
> ```
> 
> Let me know if that works.

Just a bit of update. I can finally get it works with iPython version updated. Works fine with iPython 7.15.0 now. Thanks for the responses.

**Comment by DanielGoldfarb** [comment]
Great news @moonmoondog .  Thanks for the update.

**Comment by DanielGoldfarb** [comment]
I have decided this is indeed a bug.  I was under the impression that kwarg `block` had two possible values, `True` and `False`.  Apparently it has also a third possible value: `None` which behaves differently that either `True` or `False`.  

What `None` does is **behave as if `True` for regular Python and simple Ipython, but behave as if `False` for "Ipython with its pylab mode"** (which is apparently what Spyder uses).  

Therefore I am changing the **default** value of `block` to be `None`.  Users can override the default by setting the kwarg to either `True` or `False`.  I was able to reproduce the issue running Spyder on Ubuntu, and confirmed that changing the default value fixes the issue.  

I expect to release this change by the end of this week.
